make_music_records.py

```diff
@@ -50,6 +50,10 @@ def main() -> None:
         default="INFO",
         help="Set the logging level",
     )
+    # Optional flag to include 590 field in MARC records for CDs without cases.
+    # Default is that we have cases, so no 590 field is added.
+    # argparse.BooleanOptionalAction gives us --no-cases to set the flag to False.
+    parser.add_argument("--cases", action=argparse.BooleanOptionalAction)
```

Comment:
I wasn't aware of `BooleanOptionalAction`.  It looks interesting, but unfortunately it does not seem to be useful here.  The `cases` argument is not required, and when not provided, it does not default to `True`, but to `None`.

I stuck a `print(vars(args))` after `parse_args()` and ran the script like this:
```
$ docker-compose exec batchcd python make_music_records.py -e 1 batch_016_20240229.tsv

{'music_data_file': 'batch_016_20240229.tsv', 'start_index': 0, 'end_index': 1, 'log_level': 'INFO', 'cases': None}
```

Since `cases` is `None`,  when that's passed to `add_local_fields()`, `has_cases` is `None`, so the 590 field gets added when it shouldn't (I didn't request `--no-cases`).

It does work as expected if the user explicitly runs the program with `--cases` or with `--no-cases`.  But as an optional argument with a "False-y" default, it leads to unexpected results when omitted.

If you can find a way to set a default value of `True` for `cases`, I think that's a good approach.  Otherwise, the normal way to do this is with the less-intuitive `store_true`:
```
parser.add_argument("--no-cases", help="CDs do not have cases", action="store_true")

# No argument provided, default value is False... not intuitive
$ docker-compose exec batchcd python make_music_records.py -e 1 batch_016_20240229.tsv
{'music_data_file': 'batch_016_20240229.tsv', 'start_index': 0, 'end_index': 1, 'log_level': 'INFO', 'no_cases': False}

# Argument provided
docker-compose exec batchcd python make_music_records.py -e 1 batch_016_20240229.tsv --no-cases
{'music_data_file': 'batch_016_20240229.tsv', 'start_index': 0, 'end_index': 1, 'log_level': 'INFO', 'no_cases': True}
```

The logic in `add_local_fields()` would then need to be inverted, so the 590 gets added when `no_cases == True`.